### PR TITLE
build(storybook): deploy storybook

### PR DESCRIPTION
**Pull request by andre-code**
## Description

This PR introduces the deployment of Storybook under the URL path `/storybook/`.

## Testing

You can test the Storybook deployment at the following URL:
https://renku-ci-ui-2773.dev.renku.ch/storybook/

Please review and verify that the Storybook is functioning as expected, and all UI components are accessible.

/deploy #persist

### CODE REVIEW

File: .github/workflows/acceptance-tests.yml [hunk 1]
```diff
@@ -129,6 +129,26 @@ jobs:
...
+    timeout-minutes: 60
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./client
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
```
Comment (leafty) [inline .github/workflows/acceptance-tests.yml:144]: To follow our other action and Dockerfiles:
```suggestion
          node-version: "18.17.1"
```

File: .github/workflows/acceptance-tests.yml [hunk 2]
```diff
@@ -129,6 +129,26 @@ jobs:
...
+      - name: Install dependencies
+        run: npm i
```
Comment (leafty) [inline .github/workflows/acceptance-tests.yml:146]:
```suggestion
        run: npm install
```

File: .github/workflows/acceptance-tests.yml [hunk 3]
```diff
@@ -129,6 +129,26 @@ jobs:
...
+      - name: Run Storybook tests
+        run: npm run storybook-test
+        env:
+          TARGET_URL: 'https://renku-ci-ui-${{ github.event.number }}.dev.renku.ch/storybook/'
```
Comment (leafty) [inline .github/workflows/acceptance-tests.yml:150]: I would like to try this:

```suggestion
          TARGET_URL: ${{ needs.deploy-pr.environment.url }}
```

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
I'm not sure we should rely on the PR deployment for testing Storybook since it's not strictly required (as in https://github.com/SwissDataScienceCenter/renku-ui/pull/2762) and PR deployments are rather unstable atm, so that would make this test shaky as well.

Can we remove the bits for running tests from this PR? If you need them to keep working on other issues, I suggest adding that part temporarily in a separate draft PR

File: client/package.json [hunk 4]
```diff
@@ -86,11 +86,11 @@
...
     "format-changed": "pretty-quick --pattern \"client/**/*.*\"",
     "format-commit": "pretty-quick --staged --pattern \"client/**/*.*\"",
-    "storybook": "storybook dev -p 6006 -s public",
-    "storybook-build": "storybook build -s public --quiet",
+    "storybook": "storybook dev -p 6006",
+    "storybook-build": "storybook build --quiet",
     "storybook-start-server": "http-server storybook-static --port 6006 --silent",
     "storybook-wait-server": "wait-on http://127.0.0.1:6006",
-    "storybook-test": "test-storybook --maxWorkers=2",
+    "storybook-test": "test-storybook ",
```
Comment (lorenzo-cavazzi) [inline client/package.json:93]:
```suggestion
    "storybook-test": "test-storybook",
```

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
Very nice! Just a tiny change

**Pull request merged**